src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -199,6 +204,15 @@ private static void sendNotifications(BitbucketSCMSource source, Run<?, ?> build
             listener.getLogger().println("[Bitbucket] Notifying commit build result");
             key = getBuildKey(build, r.getHead().getName(), shareBuildKeyBetweenBranchAndPR);
             bitbucket = source.buildBitbucketClient();
+            if (sourceContext.doNotOverridePrBuildStatuses()) {
```

Comment:
Should this have a special case for reporting the status of a pull request build?

Like, there is a pull request and it's first built once and the build status goes to Bitbucket all right. Then a user of Jenkins replays the build — there is no new commit. When the second build finishes, does the plugin decide not to report the status to Bitbucket because that commit already has one build status from a pull request — even though it's from the same pull request? It seems to me that the previous build status should be overwritten in that case.

Reply:
for this case it shouldn't be a problem as notification for PRs being processed in the previous `if` statement, this is for "branches" only, you could check line 198